libraries/botframework-connector/tests/auth/managedIdentityAuthenticator.test.js

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+const assert = require('assert');
+const sinon = require('sinon');
+const { ManagedIdentityAuthenticator } = require('../../lib/auth/managedIdentityAuthenticator');
+const { JwtTokenProviderFactory } = require('../../lib/auth/jwtTokenProviderFactory');
+
+const testAppId = 'foo';
+const testAudience = 'bar';
+const authResult = {
+    token: '123',
+    expiresOnTimestamp: 3000,
+};
+
+const tokenProvider = new JwtTokenProviderFactory(testAppId);
+tokenProvider.createAzureServiceTokenProvider = sinon.spy(() => ({
+    getToken: () => authResult,
+}));
+
+const testInput = [
+    { name: 'null', value: null },
+    { name: 'empty', value: '' },
+    { name: 'blank', value: ' ' },
+];
+
+const errorMsgAppId = 'ManagedIdentityAuthenticator.constructor(): missing appid.';
+const errorMsgAudience = 'ManagedIdentityAuthenticator.constructor(): missing resource.';
+const errorMsgToken = 'ManagedIdentityAuthenticator.constructor(): missing tokenProviderFactory.';
+
+describe('ManagedIdentityAuthenticator', function () {
+    describe('Constructor', function () {
+        it('should call createAzureServiceTokenProvider', function () {
+            const sut = new ManagedIdentityAuthenticator(testAppId, testAudience, tokenProvider);
+            assert.strictEqual(testAudience, sut.resource);
+            assert.strictEqual(tokenProvider.createAzureServiceTokenProvider.calledOnce, true);
+        });
+
+        testInput.forEach(({ name, value }) => {
+            it(`should throw if appId is ${name}`, function () {
+                assert.throws(() => new ManagedIdentityAuthenticator(value, testAudience, tokenProvider), {
+                    name: 'AssertionError',
+                    message: errorMsgAppId,
+                });
+            });
+        });
+
+        testInput.forEach(({ name, value }) => {
+            it(`should throw if audience is ${name}`, function () {
+                assert.throws(() => new ManagedIdentityAuthenticator(testAppId, value, tokenProvider), {
+                    name: 'AssertionError',
+                    message: errorMsgAudience,
+                });
+            });
+        });
+
+        it('should throw with null tokenProviderFactory', function () {
+            assert.throws(() => new ManagedIdentityAuthenticator(testAppId, testAudience, null), {
+                name: 'AssertionError',
+                message: errorMsgToken,
+            });
+        });
+    });
+
+    describe('getToken', function () {
+        it('should get token', async function () {
```

Comment:
I think simplifying this test suite by collapsing the `describe('getToken')` and `it('...')` would create a better dev experience for the maintainers. Example:

```ts
it('getToken() should get token', async function () {
```

Please apply this where easily applicable and use [simpleCredentialProvider.test.js](https://github.com/microsoft/botbuilder-js/blob/9c9f8d95927ea2f035af97ba53ddaa73e0231289/libraries/botframework-connector/tests/simpleCredentialProvider.test.js#L15-L21) as a reference point.